src/components/Search/CompareWithBase.tsx

```diff
@@ -24,67 +23,43 @@ interface CompareWithBaseProps {
   isEditable: boolean;
   baseRevs: Changeset[];
   newRevs: Changeset[];
-  baseRepos: Repository['name'][];
-  newRepos: Repository['name'][];
-}
-
-interface RevisionsState {
-  revs: Changeset[];
-  repos: Repository['name'][];
 }
 
 interface InProgressState {
   revs: Changeset[];
-  repos: Repository['name'][];
   isInProgress: boolean;
 }
 
 function CompareWithBase({
   isEditable,
   baseRevs,
   newRevs,
-  baseRepos,
-  newRepos,
 }: CompareWithBaseProps) {
   const [expanded, setExpanded] = useState(true);
   const { enqueueSnackbar } = useSnackbar();
 
   //the "committed" base and new revisions initialize the staging state
-  const [baseStaging, setStagingBase] = useState<RevisionsState>({
-    revs: baseRevs,
-    repos: baseRepos,
-  });
+  const [baseStagingRevs, setStagingBaseRevs] = useState<Changeset[]>(baseRevs);
 
-  const [newStaging, setStagingNew] = useState<RevisionsState>({
-    revs: newRevs,
-    repos: newRepos,
-  });
+  const [newStagingRevs, setStagingNewRevs] = useState<Changeset[]>(newRevs);
 
   //the edit button will initialize the "in progress" state
   //and copy "stage" to "in progress" state
   const [baseInProgress, setInProgressBase] = useState<InProgressState>({
     revs: [],
-    repos: [],
     isInProgress: false,
   });
```

Comment:
I wondered about splitting this in 2 different states, but finally decided to keep it like this for now.